components/cice/cime_config/buildlib

```diff
@@ -42,19 +42,33 @@ if (-f "Filepath") {
 #  force gmake to recompile if any of the cppdefs are different 
 #-------------------------------------------------------------- 
 my $recompile = 'FALSE';
-if (-e "$OBJROOT/ice/obj/CCSM_cppdefs") {
+my $sysmod = "cp -f $OBJROOT/ice/obj/CIME_cppdefs.new $OBJROOT/ice/obj/CIME_cppdefs";
+if (-e "$OBJROOT/ice/obj/CIME_cppdefs") {
+    if (compare("$OBJROOT/ice/obj/CIME_cppdefs.new", "$OBJROOT/ice/obj/CIME_cppdefs") != 0) {
+        $recompile = 'TRUE';
+    }
+    # NOTE - the following remove statements are what force cice to recompile if
+    # the cppdefs are different
+    print "recompile is $recompile \n";
+    if ($recompile eq 'TRUE') {
+        my $sysmod = "rm -f $OBJROOT/ice/obj/*.o";
+        system($sysmod) == 0 or die "ERROR cice buildlib: $sysmod failed: $?\n";
+    }
+    my $sysmod = "cp -f $OBJROOT/ice/obj/CIME_cppdefs.new $OBJROOT/ice/obj/CIME_cppdefs";
+}
+elsif (-e "$OBJROOT/ice/obj/CCSM_cppdefs") {
     if (compare("$OBJROOT/ice/obj/CCSM_cppdefs.new", "$OBJROOT/ice/obj/CCSM_cppdefs") != 0) {
```

Comment:
You should also replace CCSM_cppdefs.new with CIME_cppdefs.new everywhere.

Reply:
Ok I changed cice/cime_config/buildlib/ to store
the correct cppdef filename root in a variable, and
I also replaced CCSM_cppdefs.new with CIME_cppdefs.new everywhere.